catkin_tools/verbs/catkin_clean/clean.py

```diff
@@ -0,0 +1,201 @@
+    # Update the names of packages to be cleaned with dependants
+    packages_to_be_cleaned = determine_packages_to_be_cleaned(
+        context,
+        clean_dependants,
+        names_of_packages_to_be_cleaned)
+
+    # print(packages_to_be_cleaned)
+    # for path, pkg in packages_to_be_cleaned:
+    # if os.path.exists(os.path.join(context.build_space_abs, pkg.name)):
+    # print("[clean] Cleaning package: %s" % pkg.name)
+
+    # Construct jobs
+    jobs = []
+    for path, pkg in packages_to_be_cleaned:
+
+        # Get all build type plugins
+        clean_job_creators = {
+            ep.name: ep.load()['create_clean_job']
+            for ep in pkg_resources.iter_entry_points(group='catkin_tools.jobs')
+        }
+
+        # It's a problem if there aren't any build types available
+        if len(clean_job_creators) == 0:
+            sys.exit('Error: No build types availalbe. Please check your catkin_tools installation.')
+
+        # Determine the job parameters
+        clean_job_kwargs = dict(
+            context=context,
+            package=pkg,
+            package_path=path,
+            dependencies=[],  # Unused because clean jobs are not parallelized
+            dry_run=dry_run,
+            clean_build=True,
+            clean_devel=True,
+            clean_install=True)
+
+        # Create the job based on the build type
+        build_type = get_build_type(pkg)
+
+        if build_type in clean_job_creators:
+            jobs.append(clean_job_creators[build_type](**clean_job_kwargs))
+
+    if len(jobs) == 0:
+        print("[clean] There are no products from the given packages to clean.")
+        return False
+
+    # Queue for communicating status
+    event_queue = Queue()
+
+    # Spin up status output thread
+    status_thread = ConsoleStatusController(
+        'clean',
+        ['package', 'packages'],
+        jobs,
+        1,
+        [pkg.name for _, pkg in context.packages],
+        [p for p in context.whitelist],
+        [p for p in context.blacklist],
+        event_queue,
+        show_notifications=False,
+        show_active_status=False,
+        show_buffered_stdout=verbose or False,
+        show_buffered_stderr=True,
+        show_live_stdout=False,
+        show_live_stderr=False,
+        show_stage_events=False,
+        show_full_summary=False,
+        pre_start_time=pre_start_time,
+        active_status_rate=10.0)
+    status_thread.start()
+
+    # Initialize locks (none need to be configured here)
+    locks = {
+    }
+
+    # Block while running N jobs asynchronously
+    try:
+        ej = execute_jobs(
+            'clean',
+            jobs,
+            locks,
+            event_queue,
+            context.log_space_abs,
+            max_toplevel_jobs=1,
+            continue_on_failure=True,
+            continue_without_deps=False)
+        all_succeeded = run_until_complete(ej)
+    except Exception:
+        status_thread.keep_running = False
+        all_succeeded = False
```

Comment:
`all_succeeded` doesn't appear to be used. Should it be? If not please remove it.